barretenberg/cpp/src/barretenberg/vm/avm/trace/execution.hpp

```diff
@@ -27,18 +28,15 @@ class Execution {
     static VmPublicInputs convert_public_inputs(std::vector<FF> const& public_inputs_vec);
 
     // TODO: Clean these overloaded functions. We probably need less and confusing overloading.
-    static std::vector<Row> gen_trace(std::vector<Instruction> const& instructions,
-                                      std::vector<FF>& returndata,
+    static std::vector<Row> gen_trace(std::vector<FF>& returndata,
```

Comment:
Put the bytecode at the end of the params because it as easier than re-arranging all the calls to this function.

One that can be cleaned up by a refactor later

Reply:
What is this bytecode supposed to contain? Bytecode of 1 contract, of many?

Reply:
This is the bytecode of the top-level call that kicks off the execution. This will basically always be the dispatch function bytecode of first contract.

We end up tracking it separately since the derived address from this bytecode will be one in the public inputs